config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml

```diff
@@ -70,6 +70,8 @@ presubmits:
     always_run: false
     optional: true
     decorate: true
+    decoration_config:
+      timeout: 4h
```

Comment:
I'm not a huge fan of having a PR test that can take up to 4 hours, that would lead to a pretty bad dev experience. Have you considered other remediations that we discussed in slack like increasing parallelism or breaking up the job into 2 or 3 jobs?

Reply:
one possible solution: having the periodic job take 4 hours is ok. For the PR tests, we already have some of the tests that run in the e2e-full job as part of the e2e job that runs on every PR. Maybe we should replace the "full" PR job with an "optional" one that runs everything _but_ the tests already in the default "e2e" one. That would reduce the number of tests run and not reduce coverage.

Reply:
I like this 2nd idea. How about instead of `cluster-api-provider-azure-e2e-full` we call it `cluster-api-provider-azure-e2e-flavors`?

Do we have a preference for adding a common prefix/suffix string to each of these long cluster flavors tests, or do we want to use "GINKGO_SKIP" on the "flavors" test that include all of the cluster flavors that we have distinct tests for?

(Also feel free to nominate another alternative to "full"; I don't think we want to use "full" going forward if we're not going to actually run the full set of cluster flavor tests.)

Reply:
I like the idea of having a common prefix/suffix on each of the long flavors so we can easily SKIP/FOCUS those and not have to maintain a regex in test-infra every time we add/remove a test flavor.

for naming how about `cluster-api-provider-azure-e2e-optional-flavors`? basically trying to communicate "this is the test suite that doesn't run unless it's triggered that contains all the flavors not run in the default e2e test"

Reply:
I've updated this PR with this guidance, and submitted a capz PR that pairs with the changes:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2100

Those 2 PRs will need to be merged at more or less the same time.